src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -844,6 +873,21 @@ void QgsRelationReferenceWidget::filterChanged()
     }
   }
   filterExpression = filters.values().join( QStringLiteral( " AND " ) );
+
+  if ( mChainFilters && mComboBox->count() > 0 )
+  {
+    mNullIndex.clear();
```

Comment:
Is this required? The value will be set anyway either in `if` or in `else`, no?

Reply:
Indeed, you're right.